### ISSUE TITLE
Android Client not sending updates

### ISSUE THREAD

**Issue by vsatmydynipnet** [traccar/traccar-client-android#367]
Hi!

Thank you for this cool software. I setup an Server running on FreeBSD which worked fine in shortest time.

The IOS Client sends updates and is shown as expected.

The Android client, installed a) from F-Droid and b) via Yalp Store from Google Play does not sent updates.

- Config:
-- Samsung SM-T585 Tablet (No SIM Card, Wifi only connected)
-- Installed LineageOS 14.1 without GApps, so no Google Play Service is installed
-- Changed Url, rest default and enabled service: says Network is online.
-- Added Batterie exception on Android
-- Opened AFWAll+ Firewall for Traccar client

Server and also server log, does not show anything from this device. Everything else on this tablet is working fine and when i do a
telnet IP 5055
to my server via termUX, i do see the connection in the server log. So port is open and connectivity should be ok from this device.

Actually i run out of ideas and any hint would be appreziated.

Many thanks!

**Comment by tananaev** [comment]
Have you read Traccar Client troubleshooting guide?

**Comment by vsatmydynipnet** [comment]
- service startet
- network online
Missing;: Location update

**Comment by tananaev** [comment]
It means that your device doesn't report location. Carefully follow all troubleshooting instructions.

**Comment by vsatmydynipnet** [comment]
Installed LineageOS 14.1 without GApps, so NO Google Play Service is installed

Without Gapps should work ok?

**Comment by tananaev** [comment]
Yes, it should work just fine without Google Apps.

**Comment by vsatmydynipnet** [comment]
Ok, will recheck everything and report back. Thank you!

**Comment by vsatmydynipnet** [comment]
Rechecked everything, incl. deactivation of firewall, aso.
Location Service is running, all permissions set
com.location.android.fused running

I am out of ideas.

**Comment by vsatmydynipnet** [comment]
I changed from energy saving mode to high accuracy on android system settings & traccar client, used gps status to check gps is working and it reports immediately. it does not report with low and medium on traccar side, even gps is on. ButI assume GPS/ high is a battery drainer.

**Comment by vsatmydynipnet** [comment]
Sir, please read my posts to see I have checked everything.  am not an android Specialist, but I wrote I tested with Browser and Telnet from the device. Furthermore I wrote:

Rechecked everything, incl. deactivation of firewall, aso.
Location Service is running, all permissions set
com.location.android.fused running

Actually I found gpsstatus to check GPS is running. I have not found an App to check Low/Medium use. Any suggestions?

**Comment by vsatmydynipnet** [comment]
All written above.  
Android Location service is on, 3 Modes in Location settings GPS, WLAN..., GPS WLAN.... Testet all 3, Te WLAN/Network one is not working. As long as GPS is included, it works IF set to HIGH on Traccar client.

I don't install Google Maps App, but will check if there is something else.

**Comment by tananaev** [comment]
Sounds like network provider just doesn't work on your phone. It depends on Google service, so maybe you got it blocked somehow.

**Comment by oliv3** [comment]
Hi @vsatmydynipnet , I will try to lend a hand:
 * I heavily use [SatStat](https://github.com/mvglasow/satstat) (not only for debugging, but also because it's an excellent software, and the UI is just pretty).
 * You might want to give a try to https://github.com/traccar/traccar-client-android/compare/master...oliv3:locationProvider. This branch adds analytics that may help you finding out what's wrong.
 * I can provide you a .apk if you can't build one, please feel free to contact me either privately or better, on IRC (https://github.com/traccar/traccar/issues/3475).

**Comment by oliv3** [comment]
@tananaev What I can confirm is that, _even with all location providers enabled on the phone_ (which, on a Sony, is GPS+Network+Bluetooth+Cellular), the client won't send anything _unless_ you use the "high accuracy" setting. "medium" and "low" do not work, even with the device moving.
HTH.

**Comment by vsatmydynipnet** [comment]
here it works using GPS+Network+Bluetooth+Cellular or GPS device only on Android and High on Traccar. The Energy saving WLAN... mode doesn't work. Access to Google is not blocked here, but as mentioned above, no Google apps or services installed on the device. Naked LineageOS only.

And also written above I connected it via my mobile phone too, so mobile is connected via LTE, creating an hotspot for the device. This way all things are open.

Took my IPhone with me for bicycle tour. Reporting on IOS is correct, medium reports every 5 minutes, so tour is recorded fine. Traccar uses far more battery on IOS then Owntracks. This is kinda problem.

**Comment by oliv3** [comment]
I can confirm that with GPS+Network+Cellular+Bluetooth, it only works with the "high" setting.

**Comment by oliv3** [comment]
(Unfortunately I don't use Apple products).

**Comment by vsatmydynipnet** [comment]
I am mid in moving away all devices and with LineageOS a secure and bloatware free device, respecting the privacy is possible. My last problem was replacing owntracks with something working without installed Gapps on it. I think we close this for now.

I will buy an additional phone, which is fully supported by LineageOS and retry it with this one. The actual one is some tablet without SIM card. Will report again, after install and test is done.

Thank you very much and enjoy the sunday!

**Comment by oliv3** [comment]
Have you looked at https://e.foundation/ ?

**Comment by vsatmydynipnet** [comment]
Looks like they try to start something as Puri.sm with their Librem5. Hopefully alot of these types comes up. Reads promising

**Comment by garyvdm** [comment]
Guys - please move this discussion to the forums.

**Comment by oliv3** [comment]
Not sure if the forums are the best place for that.

**Comment by oliv3** [comment]
@tananaev OK, experiments here confirm that _IF_ GPS is turned on the phone, and Traccar uses the "High" accuracy setting, location updates occur.
When using "Medium" or "Low" setting, Traccar will not send locations _UNLESS_ another application uses GPS.
I expect this to be worse when relying only on Network. 
This with a Sony BTW. So I don't really expect some kind of weird manufacturer tuning.
Can anyone confirm ?

**Comment by oliv3** [comment]
And (absolutely not tested, though) I would also suspect the Hidden application flavor to be really stealth in that case.

**Comment by vsatmydynipnet** [comment]
Morning!  @oliv3 This is same here on Samsung SM-T585(Galaxy Tab A (2016)) Tablet.
I have not installed the  Hidden App, socan't confirm for this one.

**Comment by greatidea1** [comment]
I agree with above comments, as if my below experience is helpful for anybody.

I'm using Traccar client on Android 6, GPS on high accuracy, app allowed in firewall and phone left in high speed 4G connection in open sky area, works for a while, then suddenly stops sending updates. 
I observed initially the app's icon is displayed at the top status bar, but when it disappears suddenly, it is declared as offline in Traccar web. 
Another observation: When I explicitly open the client app again, it starts transmitting, so the app quits  or is killed by the Android OS. I also checked, my phone doesn't have memory management and is not set to kill apps automatically.

**Comment by tananaev** [comment]
Are you sure your phone doesn't have battery optimisation (e.g. Doze mode)?

**Comment by oliv3** [comment]
@tananaev Client still doesn't send anything **unless** accuracy setting is set to "High" **and** GPS is activated on the phone, _or_ accuracy setting is "Medium" or "Low" **and** another application is requesting location updates.
Cellular/Wi-Fi network is basically **not** used, **even** with "High" setting.
Any hint appreciated, thanks !

**Comment by oliv3** [comment]
I really have no idea. I never changed those parts.

**Comment by tananaev** [comment]
You can debug and check provider source.

**Comment by oliv3** [comment]
Maybe you should try to revert back to MixedLocationProvider that was before using Lost ? Then again, I have no clue. You are the Android expert.

**Comment by oliv3** [comment]
My patch sending the provider along other parameters just reports "fused". All the time.

**Comment by tananaev** [comment]
I have decided to make another change to explicitly select location providers based on accuracy selected in the app (see ac6d677ff45d3581c1b4ea16f41cb517a21d8398).

**Comment by oliv3** [comment]
Has 5.15 been thoroughly tested ?

**Comment by tananaev** [comment]
I have tested in on emulator and one device. Is it not working for you?

**Comment by oliv3** [comment]
I have not had time to test it yet. But since https://github.com/traccar/traccar-client-android/commit/ac6d677ff45d3581c1b4ea16f41cb517a21d8398 and https://github.com/traccar/traccar-client-android/releases/tag/v5.15 happened just 8 hours ago, I was wondering.

**Comment by tananaev** [comment]
I did some testing, but if something is not working, please let me know.

**Comment by oliv3** [comment]
I'm pretty busy working on other projects, can't guarantee anything.

**Comment by oliv3** [comment]
Hopefully @vsatmydynipnet, @garyvdm or @greatidea1 could help ?

**Comment by tananaev** [comment]
Sure, that's ok.

**Comment by n2yo** [comment]
I am having a hard time with the hidden app to get location from network provider (when GPS signal is unavailable. An extended explanation was submitted here: https://www.traccar.org/forums/topic/issues-with-location-network-provider/
I am not Android programmer, but I wonder if any of those 2 things may play a role in this problem that occurs on some phones:
1) The Android bug reported here: https://issuetracker.google.com/issues/36975498
2) The requirement. if Android is 5.0 (API level 21) or higher, that app must declare android.hardware.location.network or android.hardware.location.gps hardware feature in the manifest file
Details here https://developer.android.com/guide/topics/location/strategies
I didnt see that in the AndroidManifest.xml

**Comment by tananaev** [comment]
I have just released version 5.16 that addresses your second point, but I don't think it would change the behaviour of the app.

**Comment by n2yo** [comment]
After I posted my earlier comment I compiled myself 5.15 with the recommended changes in AndroidManifest.xml then installed a debug version on the phone with problems. It worked, and still works. But I saw this behavior before, the app may stop working hours or days later. So I will leave it working for a couple of days before draw any conclusions.
